x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -298,22 +356,37 @@ export class SearchSessionService
 
   public asScopedProvider = ({ savedObjects }: CoreStart) => {
     return (request: KibanaRequest) => {
+      const user = this.security?.authc.getCurrentUser(request) ?? null;
       const savedObjectsClient = savedObjects.getScopedClient(request, {
         includedHiddenTypes: [SEARCH_SESSION_TYPE],
       });
       const deps = { savedObjectsClient };
       return {
-        getId: this.getId.bind(this, deps),
-        trackId: this.trackId.bind(this, deps),
-        getSearchIdMapping: this.getSearchIdMapping.bind(this, deps),
-        save: this.save.bind(this, deps),
-        get: this.get.bind(this, deps),
-        find: this.find.bind(this, deps),
-        update: this.update.bind(this, deps),
-        extend: this.extend.bind(this, deps),
-        cancel: this.cancel.bind(this, deps),
-        delete: this.delete.bind(this, deps),
+        getId: this.getId.bind(this, deps, user),
+        trackId: this.trackId.bind(this, deps, user),
+        getSearchIdMapping: this.getSearchIdMapping.bind(this, deps, user),
+        save: this.save.bind(this, deps, user),
+        get: this.get.bind(this, deps, user),
+        find: this.find.bind(this, deps, user),
+        update: this.update.bind(this, deps, user),
+        extend: this.extend.bind(this, deps, user),
+        cancel: this.cancel.bind(this, deps, user),
+        delete: this.delete.bind(this, deps, user),
       };
     };
   };
+
+  private throwOnUserConflict = (
+    user: AuthenticatedUser | null,
+    session?: SavedObject<SearchSessionSavedObjectAttributes>
+  ) => {
+    if (user === null || !session) return;
+    if (
+      user.authentication_realm.type !== session.attributes.realmType ||
+      user.authentication_realm.name !== session.attributes.realmName ||
+      user.username !== session.attributes.username
+    ) {
+      throw notFound();
```

Comment:
I decided to remove the `throwOnUserConflict` portion, since it's unlikely a user will actually attempt to retrieve another user's session (unless they use the APIs directly). It also resulted in many of the extra calls you mentioned above.